clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/util/id/DcosSpinnakerAppId.groovy

```diff
@@ -0,0 +1,153 @@
+package com.netflix.spinnaker.clouddriver.dcos.deploy.util.id
+
+import static com.google.common.base.Preconditions.checkArgument
+import static com.google.common.base.Strings.nullToEmpty
+
+import com.netflix.frigga.Names
+
+/**
+ * Represents a hierarchical Spinnaker specific application identifier for DCOS.
+ * Structure - /account/region/app-stack-detail-sequence
+ */
+class DcosSpinnakerAppId {
+    public final static def SAFE_REGION_SEPARATOR = "_"
+
+    private final def marathonPath
+
+    public DcosSpinnakerAppId(final String marathonAppId) {
```

Comment:
> From just the constructor, you don't get that information.

Right, I get that, but you don't really need that context since they can create the exact same objects. I don't think that extra context is necessary. I will make the change regardless because I have to to avoid parameter issues regardless, I just don't think it would be useful outside of that.

I will move both to static factory methods though, since I as well prefer consistency with the entry point for using an object.